lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -3736,29 +3737,58 @@ estimateSignedTxSizeSpec = describe "estimateSignedTxSize" $ do
         -> ByteString
         -> Cardano.Tx era
         -> IO ()
-    test _name bs tx@(Cardano.Tx (body :: Cardano.TxBody era) _) = do
+    test _name bs cardanoTx@(Cardano.Tx body _) = do
         let pparams = Write.pparamsLedger $ mockPParamsForBalancing @era
-            utxo = utxoPromisingInputsHaveVkPaymentCreds body
-            witCount = estimateKeyWitnessCount (Write.fromCardanoUTxO utxo) body
+            witCount dummyAddr = estimateKeyWitnessCount
+                    (Write.fromCardanoUTxO
+                        $ utxoPromisingInputsHaveAddress dummyAddr body)
+                    body
```

Comment:
Could consider flipping the arguments to avoid the awkward multi-line parentheses:
```suggestion
            witCount dummyAddr
                = flip estimateKeyWitnessCount body
                $ Write.fromCardanoUTxO
                $ utxoPromisingInputsHaveAddress dummyAddr body
```

Reply:
I believe I prefer it the way it is. I don't mind the parenthesis too much, but find the flipping disorienting.